frontend/packages/react-app/src/components/organisms/TokenCampaignDetail/index.tsx

```diff
@@ -17,60 +17,102 @@
+  useEffect(() => {
+    switch (campaign.distributor.type) {
+      case "wallet":
+        setDistributorType("Wallet Address Type");
+        break;
+      case "uuid":
+        setDistributorType("UUID Type");
+        break;
+      case "email":
+        setDistributorType("Email Address Type");
+        break;
+      default:
+        setDistributorType(campaign.distributor.type);
+    }
+  }, [campaign]);
 
   return (
-    <div style={{ marginTop: "24px" }}>
-      <Paper className={classes.container}>
+    <>
+      <div style={{ padding: theme.spacing(4) }}>
         <Box display="flex" justifyContent="space-between">
           <Typography variant="h4" component="h2">
             {campaign.campaignMetadata.name}
           </Typography>
           <CampaignStatusChip status={campaign.status} />
         </Box>
         <Box mt={2}>
-          <Typography variant="subtitle1">Description:</Typography>
+          <Typography variant="caption" style={{ color: "#797979" }}>
+            Description:
+          </Typography>
           <Typography variant="body1">
             {campaign.campaignMetadata.description !== ""
               ? campaign.campaignMetadata.description
               : "No description"}
           </Typography>
         </Box>
-        <Box display="flex" justifyContent="start" mt={2}>
-          <Box>
-            <Typography variant="subtitle1">Start Date:</Typography>
-            <Typography variant="h5">
+        <Box display="flex" justifyContent="start" mt={1} mb={1}>
+          <Box display="flex" justifyContent="start" alignItems="baseline">
+            <Typography variant="caption" style={{ color: "#797979" }}>
+              Start Date:
+            </Typography>
+            <Typography variant="body1">
               {new Date(
                 parseInt(campaign.startDate) * 1000
               ).toLocaleDateString()}
             </Typography>
           </Box>
-          <Box ml={8}>
-            <Typography variant="subtitle1">End Date:</Typography>
-            <Typography variant="h5">
+          <Box
+            ml={4}
+            display="flex"
+            justifyContent="start"
+            alignItems="baseline"
+          >
+            <Typography variant="caption" style={{ color: "#797979" }}>
+              End Date:
+            </Typography>
+            <Typography variant="body1">
               {new Date(parseInt(campaign.endDate) * 1000).toLocaleDateString()}
             </Typography>
           </Box>
         </Box>
-      </Paper>
-    </div>
+        <Box display="flex" justifyContent="start" alignItems="baseline">
+          <Typography variant="caption" style={{ color: "#797979" }}>
+            Distributor Type:
+          </Typography>
+          <Typography variant="body1" style={{ paddingLeft: 8 }}>
+            {distributorType}
+          </Typography>
+        </Box>
+        <Box>
+          {/* //TODO: ADD LINK */}
+          <Link
+            href="#"
```

Comment:
This one is better than no contents? 
https://medium.com/iroiro-social-token/introducing-how-to-distribute-token-by-wallet-address-or-url-4e4646581047

Reply:
I was looking for that!